### ISSUE TITLE
didi types cause errors in build

### ISSUE THREAD

**Issue by aapoalas** [nikku/didi#11]
__Describe the Bug__

In a project using `didi` for dependency injection, the new version with types included causes TS errors. It seems that the errors are so terrible to TypeScript that it actually will actually refuse to ignore the errors even with `node_modules` excluded.

The errors are the following three:
```
node_modules/didi/dist/types/lib/types.d.ts:56:3 - error TS2411: Property '__exports__' of type 'string[] | undefined' is not assignable to string index type 'string[] | ModuleDeclaration[] | ["value", any] | ["value", any, "private"] | ["type", TypeDeclaration<any>] | ["type", TypeDeclaration<any>, "private"] | [...] | [...]'.

56   __exports__?: string[],
     ~~~~~~~~~~~

node_modules/didi/dist/types/lib/types.d.ts:57:3 - error TS2411: Property '__modules__' of type 'ModuleDeclaration[] | undefined' is not assignable to string index type 'string[] | ModuleDeclaration[] | ["value", any] | ["value", any, "private"] | ["type", TypeDeclaration<any>] | ["type", TypeDeclaration<any>, "private"] | [...] | [...]'.

57   __modules__?: ModuleDeclaration[],
     ~~~~~~~~~~~

node_modules/didi/dist/types/lib/types.d.ts:58:38 - error TS8020: JSDoc types can only be used inside documentation comments.

58   [name: string]: ServiceDeclaration<?> | ModuleDeclaration[] | string[]
```

__Steps to Reproduce__

1. Setup a new project
2. Install `didi@6.0.0`
3. Setup a tsconfig with `"module": "es2020"` and `"strict": true` (possibly `"target": "es2020"` and `"moduleResolution": "node"` as well). Exclude node_modules.
4. Create a `main.ts` where "didi" is dynamically imported with `const { Injector } = await import("didi");`.
5. Run tsc build
6. Errors should appear in build.

__Expected Behavior__

Library TS definitions should not contain errors.

It seems that
1. The ModuleDeclaration arrays should not be optional.
2. The `?` in the `ServiceDeclaration`'s type parameter seems to be an obvious JSDoc oversight.

__Environment__

- Ubuntu 18.04 WSL2 env

**Comment by nikku** [comment]
I'll look into the issue once I find the time to do so.

In the meantime, please use [`v5.x`](https://github.com/nikku/didi/blob/master/CHANGELOG.md#521) that ships without types.

**Comment by nikku** [comment]
Should be fixed with `v6.1.0`. If you find further issues, please report them.
